Added RewriteRule to redirect SSUI requests for /self_service/* to /self_service

gtanzillo opened the pull request:
The code change redirects /self_service/\* to /self_service. It, however, excludes 
- /self_service/assets
- /self_service/images
- /self_service/img
- /self_service/styles
- /self_service/js
- /self_service/fonts
  Those URLs need to be serviced as is by the SSUI app.

This addresses a 404 page when refreshing the SSUI login page.

https://bugzilla.redhat.com/show_bug.cgi?id=1275679

gtanzillo committed:
Added RewriteRule to redirect SSUI requests for /self_service/login to /self_service

https://bugzilla.redhat.com/show_bug.cgi?id=1275679

gtanzillo commented:
@Fryguy Please review

kbrock commented on COPY/etc/httpd/conf.d/cfme-https-application.conf:
```diff
@@ -38,8 +38,9 @@ SSLCertificateKeyFile /var/www/miq/vmdb/certs/server.cer.key
 <ifmodule mod_rewrite.c>
-  RewriteCond %{REQUEST_URI} ^/favicon\.ico$ [NC]
-  RewriteRule (.*) /assets/favicon.ico       [L,R=301]
+  RewriteCond %{REQUEST_URI} ^/favicon\.ico$        [NC]
```
why is this a compound rule?
would this work?

```
RewriteRule ^/favicon\.ico$ /assets/favicon.ico [L,R=301]
```

kbrock replied:
sorry - that is unrelated. merge without edit if you want. Probably don't need to reformat these lines.

gtanzillo replied:
Thanks @kbrock Yes, I just changed the spacing to line up the three entries. I'll need to update this PR anyway to handle additional URLs. Will do that shortly

gtanzillo committed:
Added RewriteRule to redirect SSUI requests to /self_service

Updated the rewrite rule to be exclusive of assets, images, img, styles, fonts and js and redirect all other requests except matching /self_service/* to self_service.

https://bugzilla.redhat.com/show_bug.cgi?id=1275679

kbrock commented:
Wow. That is a lot of look a heads. Don't know the performance characteristics. Can you just do those together with a `|` (or)?

Sorry for vague fud

gtanzillo commented:
@kbrock Those look aheads are actually and'ed together. I wanted regex to contain the list of URL suffixes that should be excluded from the redirect instead of the ones that should be included because the include list is open ended. This way, the SSUI guys can add more URLs without us having to update this configuration. If there's a better way to do this I'm open to changing it. Thanks!

kbrock commented:
Think we don't want the extra capture groups and `.+$` - just eats memory / time.

```
RewriteRule ^/favicon\.ico$  /assets/favicon.ico       [L,R=301]
RewriteRule ^/self_service/(?!(assets|images|img|styles|js|fonts)). /self_service [R=301]
```

NOTE: I tested these in ruby not in apache.

gtanzillo committed:
Added RewriteRule to redirect SSUI requests to /self_service

Simplified regexp

https://bugzilla.redhat.com/show_bug.cgi?id=1275679

gtanzillo commented:
@kbrock That's much nicer. Thanks for the help on this one!
